pkg/controllers/nodeclaim/garbagecollection/suite_test.go

```diff
@@ -332,3 +316,77 @@ var _ = Describe("GarbageCollection", func() {
 		})
 	})
 })
+
+var _ = Describe("NetworkInterface Garbage Collection", func() {
+	It("should not delete a network interface if a nodeclaim exists for it", func() {
+		// Create and apply a NodeClaim that references this NIC
+		nodeClaim := coretest.NodeClaim()
+		ExpectApplied(ctx, env.Client, nodeClaim)
+
+		// Create a managed NIC
+		nic := test.Interface(test.InterfaceOptions{Name: instance.GenerateResourceName(nodeClaim.Name), NodepoolName: nodePool.Name})
```

Comment:
Yes I suggested elsewhere to collapse these for redability, but I meant (and included in the example) only `test.Interface(test.InterfaceOptions{` part, like this:

```suggestion
		nic := test.Interface(test.InterfaceOptions{
			Name: instance.GenerateResourceName(nodeClaim.Name),
			NodepoolName: nodePool.Name,
		})
```

This is the pattern that we use in many other places for readability. The rationale is that the type names are usually boilerplate, while the field values are usually of interest.